privacy/bolton/models.py

```diff
@@ -0,0 +1,302 @@
+# Copyright 2018, The TensorFlow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Bolton model for bolton method of differentially private ML"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import tensorflow as tf
+from tensorflow.python.keras.models import Model
+from tensorflow.python.keras import optimizers
+from tensorflow.python.framework import ops as _ops
+from privacy.bolton.losses import StrongConvexMixin
+from privacy.bolton.optimizers import Bolton
+
+
+class BoltonModel(Model):
+  """
+  Bolton episilon-delta model
```

Comment:
Here is the biggest comment in my review: Somehow you mentioned here you want to implement "epsilon-delta" differential privacy, but from my read below, you only implement the case for delta = 0 (that is \eps-DP).

\eps-DP and (\eps, \delta)-DP are qualitatively different and they use different mechanisms to sample noise (see https://github.com/frankmcsherry/blog/blob/master/posts/2017-02-08.md). So in general, it would be nice to have both (indeed -- for convex learning it is fortunate that we can have both with reasonable accuracy).

My question is: What's your plan? Do you plan to leave it with eps-DP, or you plan to implement (\eps, \delta)-DP later? (not required in this PR though). I want to learn your thoughts about this.